docs/migrations.md

```diff
@@ -0,0 +1,114 @@
+
+## Q server migration endpoint
+- __GET__ _/admin/migration/{tool}/{id?}_: Endpoint to migrate all Q items of the specified `tool` or, if the `id` is given, a single item with that id. The reponse contains all item ids aggregated by their migration statuses `updated`, `not updated` and `failed`. For each item the migration endpoint of the respective tool is called which responds with either the changed item or with a `Not modified`. If an item was changed Q server will save the updated item in the database.
+
+## Database view
+The migration endpoint in Q server expects a specific view called `byTool` which you have to add to your Q item database(s):
+```json
+{
+  "_id": "_design/items",
+  "views": {
+    "byTool": {
+      "map": "function (doc) {\n  if (doc.tool) {\n    emit(doc.tool, 1);\n  }\n}",
+      "reduce": "_count"
+    }
+  }
+}
+```
+Q server uses this view without the reduce option. See also [Installation](install.html) to get an overview of all views you should define on your Q item database(s).
+
+## Migration mechanism in tools
+
+### Endpoint
+You have to define the following endpoint if you want to be able to migrate items via Q server:
+- __POST__ _/migration_: The item will be POSTed to that endpoint and the handler method runs all necessary scripts for the respective version of your tool on that item and returns either the modified item or a `Not modified` if the item was not affected by the change. 
+
+   One example of one of our election tools:
+   
+   ```javascript
+
+    const Joi = require('joi');
+    const Boom = require('boom');
+
+    // register migration scripts here in order of version, 
+    // i.e. list the smallest version first
+    const migrationScripts = [
+      require('../migration-scripts/to-v2.0.0.js')
+    ]
+
+    module.exports = {
+      method: 'POST',
+      path:'/migration',
+      config: {
+        validate: {
+          payload: {
+            item: Joi.object().required()
+          }
+        }
+      },
+      handler: (request, reply) => {
+        let item = request.payload.item;
+        const results = migrationScripts.map(script => {
+          const result = script.migrate(item);
+          if (result.isChanged) {
+            item = result.item;
+          }
+          return result;
+        })
+        const isChanged = results.findIndex(result => {
+          return result.isChanged;
+        });
+        if (isChanged >= 0) {
+          return reply({
+            item: item
+          })
+        }
+        return reply().code(304);
+      }
+    }
+   ```
+We simply execute all migration scripts, that's why it's important that migration scripts are sorted by version. It's expected that each script exports a module named `migrate` which gets an item as parameter. Return value of each migrate module is an object containing a flag whether the item was changed and the (modified) item itself.
```

Comment:
sorted -> ordered 👍 
No, the item is returned, otherwise we would not get the changes, hum?